### PR TITLE
Refactor game load functions, resolves #6011

### LINKED ISSUE #6011: Too many disparate 'load this file and play it' functions
There are far too many functions in too many places that handle loading saves or scenarios and playing them. As things need to be done in a particular order at load time it would be far easier if there were one or two that actually handle the core logic of loading, and those were called from the other locations. The signatures and naming of these functions also need cleaning up.

### PR DESCRIPTION
**Pull request by rwjuk**
This PR refactors the various disparate 'load this file and run it' functions - there is now one overarching 'load and play' function - `Context->LoadParkFromFile()`, and a C wrapper `context_load_park_from_file()` that autodetects the format and calls the appropriate lower-level functions.

I've also renamed the ParkImporter C wrappers to follow a standard pattern - `load_from_EXT`. This way it is clearer what does what, and it'll be easy to extend this with a `load_from_orct2park` (or whatever we'll call our own file format).

(I'm aware that the editor still relies on the lower-level functions, and that there may be more scope for streamlining withing Context.cpp, but this is the first step at least.)

### CODE REVIEW

**Comment by IntelOrca** [comment]
I dislike the `loadTitleScreenOnFail` parameter, however I feel like we can't remove it until we can load a park in full isolation from the current running park.

**Comment by rwjuk** [comment]
@IntelOrca I don't like it either. I'll have another look to see if I can get rid of it sooner rather than later.

**Comment by rwjuk** [comment]
Well, I've managed to get rid of the parameter, but I'm not entirely sure I like *that* solution either! If we feel it's better to just keep the parameter until such time that we can decouple park loading from the current park I'll just nuke the last commit.

**Comment by Gymnasiast** [comment]
I'd leave out the latest commit then, IntelOrca was working on that anyway.

**Comment by rwjuk** [comment]
@Gymnasiast Ahh, okay - done.

File: src/openrct2/Context.cpp [hunk 1]
```diff
@@ -707,20 +712,34 @@ namespace OpenRCT2
                         gLastAutoSaveUpdate = AUTOSAVE_PAUSE;
                         if (info.Type == FILE_TYPE::SAVED_GAME)
                         {
+                            if (network_get_mode() == NETWORK_MODE_CLIENT) {
```
Comment (Gymnasiast) [inline src/openrct2/Context.cpp]: Braces on their own lines.

File: src/openrct2/rct2/S6Importer.cpp [hunk 2]
```diff
@@ -670,7 +670,7 @@ IParkImporter * ParkImporter::CreateS6(IObjectRepository * objectRepository, IOb
 
 extern "C"
 {
-    ParkLoadResult * game_load_sv6_path(const char * path)
+    ParkLoadResult * load_from_sv6(const char * path)
```
Comment (Gymnasiast) [inline src/openrct2/rct2/S6Importer.cpp]: Wouldn't it be better to make these functions proper C++ functions and have the extern C functions only call the real ones?

Comment (rwjuk) [inline src/openrct2/rct2/S6Importer.cpp]: Yeah, might as well do that now while we're fixing other stuff.

Comment (rwjuk) [inline src/openrct2/rct2/S6Importer.cpp]: On second thought - if I make the `load_from_sxx()` functions part of the ParkImporter interface then I'll be adding two separate functions (load save & load scenario) to said interface.

Given that our custom save format won't have this distinction I wasn't sure whether it was a good idea after all. Of course, we could just have the interface implementation for our format work in such a way that one function just calls the other but...

**Pull request merged**